recovery.go

```diff
@@ -0,0 +1,70 @@
+package raft
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// PeersJSONRecovery is a recovery manager that reads the old-style peers.json
+// file. It does not support server IDs, so it should only be used along with
+// ProtocolVersion 0. Upon disarming, it'll delete the peers.json file.
+type PeersJSONRecovery struct {
+	// path is the full path to the peers.json file.
+	path string
+
+	// Configuration is the override configuration.
+	Configuration Configuration
+}
+
+// NewPeersJSONRecovery takes the given base directory and parses the peers.json
+// file at that location, returning a recovery manager to apply this
+// configuration at startup. If there's no recovery defined then this will return
+// nil, which is a valid thing to pass to NewRaft().
+func NewPeersJSONRecovery(base string) (*PeersJSONRecovery, error) {
```

Comment:
A "recovery manager" doesn't seem to really be a thing, and I don't think it needs to be. How about simply:

    func ReadPeersJSON(path) (Configuration, error)

The caller would have the burden of `path := filepath.Join(base, "peers.json")` and `os.Remove(path)`, which I think would be entirely tolerable.